core/src/main/java/org/elasticsearch/index/query/FilteredQueryBuilder.java

```diff
@@ -40,6 +46,15 @@
     static final FilteredQueryBuilder PROTOTYPE = new FilteredQueryBuilder(null, null);
 
     /**
+     * A query that applies a filter to the results of a match_all query.
+     * @param filterBuilder The filter to apply on the query (Can be null)
+     * */
+    public FilteredQueryBuilder(@Nullable QueryBuilder filterBuilder) {
+        this.queryBuilder = new MatchAllQueryBuilder();
+        this.filterBuilder = filterBuilder;
+    }
+
```

Comment:
Happy to do this - should this be mentioned in the migration_query_refactoring doc as before null was allowed here?